test/Test.hs

```diff
@@ -23,11 +22,19 @@ main = do
 
     if result
         then putStrLn "Success"
-        else putStrLn "Fail" >> exitFailure
+        else putStrLn "Some tests failed" >> exitFailure
   where
     compareWithGolden :: FilePath -> IO Bool
     compareWithGolden f = do
         let (name, _) = splitExtension f
         goldenContent <- readFile $ name <.> "golden"
         testContent   <- readFile f
-        pure $ goldenContent == testContent
+        if goldenContent == testContent
+          then return True
+          else outputFailure name testContent goldenContent >> return False
```

Comment:
Alternatively, here you can use [`<$`](https://hackage.haskell.org/package/base-4.12.0.0/docs/Prelude.html#v:-60--36-) function from `Functor` and write this `if` branch in the following way:

```haskell
else False <$ outputFailure name testContent goldenContent
```